cmd/metal-api/internal/metal/machine.go

```diff
@@ -147,6 +148,99 @@ type MachineAllocation struct {
 	MachineSetup     *MachineSetup     `rethinkdb:"setup" json:"setup"`
 	Role             Role              `rethinkdb:"role" json:"role"`
 	VPN              *MachineVPN       `rethinkdb:"vpn" json:"vpn"`
+	Egress           []EgressRule      `rethinkdb:"egress" json:"egress"`
+	Ingress          []IngressRule     `rethinkdb:"ingress" json:"ingress"`
+}
+
+type EgressRule struct {
+	Protocol Protocol `rethinkdb:"protocol" json:"protocol"`
+	Ports    []int    `rethinkdb:"ports" json:"ports"`
```

Comment:
should be []Port which is this struct:

```go
type Port struct {
     Port IntOrString // stolen from kubernetes
     Name string
}
```

otherwise specifying port ranges are hard to specify

Reply:
Not sure if this is worth the effort though

Reply:
At least in our current use-cases there is no specific need for port ranges.

Reply:
Yes but OTOH changing this in the future is hard